doc/May-2022-Upgrade.md

```diff
@@ -0,0 +1,634 @@
+        1. Restart MongoDB in standalone (remove `--replSet replicaSet1 --keyFile /keyFile` from the `entrypoint.sh` file)
+        1. Create an admin user on MongoDB `local` db (db used for replicaSets)
+        
+            ```bash  
+            user@computer:kobo-install$ ./run.py -cb exec mongo bash
+            root@mongo:/$ mongo -u root -p "$MONGO_INITDB_ROOT_PASSWORD" admin
+            > db.createUser(
+               {
+                  user: "localRoot",
+                  pwd: "<randomPassword>",
+                  roles: [ { role: "dbAdmin", db: "local" } ]
+               }
+            )
+            > exit
+            root@mongo:/$ mongo -u localRoot -p <randomPassword> admin
+            > use local;
+            > db.dropDatabase();
+            > exit
+            root@mongo:/$ mongo -u root -p "$MONGO_INITDB_ROOT_PASSWORD" admin
+            > db.dropUser('localRoot');
+            ```
+           
+    **You can now run the command described above to validate your MongoDB isntance is using `WiredTiger`.**      
+       
+1. Upgrade to 3.6
+
+    1. Stop `mongo` container
+
+        ```shell
+        user@computer:kobo-install$ python run.py --cb stop mongo  
+        ```
+       
+    1. Edit composer file `docker-compose.primary.backend.template.yml` and change image to `mongo:3.6`
+
+        ```
+        mongo:
+          image: mongo:3.6
+        ```
+   
+    1. Start the container: 
+        
+        ```shell
+        user@computer:kobo-install$ python run.py --cb up --force-recreate mongo  
+           ```
+           
+    1. Wait for MongoDB to be ready. You should see in the console the output below: 
+
+        ```
+        mongo_1        | {
+        mongo_1        |     "numIndexesBefore" : 3,
+        mongo_1        |     "numIndexesAfter" : 3,
+        mongo_1        |     "note" : "all indexes already exist",
+        mongo_1        |     "ok" : 1
+        mongo_1        | }
+        ```
+
+    1. From another terminal, enter the container and update compatibility version.
+
+        ```shell
+        root@mongo:/# mongo -u "$MONGO_INITDB_ROOT_USERNAME" -p "$MONGO_INITDB_ROOT_PASSWORD" admin
+        > db.adminCommand( { setFeatureCompatibilityVersion: "3.6" } )
+        { "ok" : 1 }
+        > exit
+        bye
+        root@mongo:/# exit
+        ```
+        
+1. Upgrade to 4.0, 4.2, 4.4 and 5.0
+
+    Repeat step above for each version and replace the version accordingly.
+    You **must** upgrade each version one by one.
+    
+    Then start the container:
+     
+    ```shell
+     user@computer:kobo-install$ python run.py --cb up -d --force-recreate mongo     
+    ```
+
+    Done!
+
+
+## Tests
+
+1. Test if upgrade is successful
+
+    Start your containers as usual.
+
+    ```shell
+    user@computer:kobo-install$ python run.py  
+    ```
+
+    Log into one of your user accounts and validate everything is working as expected.         
+
+1. Clean up
```

Comment:
If `## Tests` is an H2, then perhaps this should also be `## Clean up` / `## Cleaning up`

Reply:
It's a mistake from copy/paste from notes.